lumicks/pylake/detail/confocal.py

```diff
@@ -314,7 +370,17 @@ def save_tiff(self, filename, dtype=np.float32, clip=False):
         """
         if self.get_image("rgb").size > 0:
             save_tiff(
-                self.get_image("rgb"), filename, dtype, clip, ImageMetadata.from_dataset(self._json)
+                self.get_image("rgb"),
+                filename,
+                dtype,
+                clip,
+                ImageMetadata(
+                    pixel_size_x=self.pixelsize_um[0] * 1000,  # um => nm
+                    pixel_size_y=self.pixelsize_um[1] * 1000
+                    if len(self.pixelsize_um) > 1
+                    else None,
+                    pixel_time=self.pixel_time_seconds,
+                ),
```

Comment:
I don't think it can. It relies on `pixelsize_um` rather than the raw image metadata. `pixelsize_um` could have been affected by downsampling (or perhaps other things in the future). I do agree that this `ImageMetaData` class feels a bit superfluous though (and potentially easy to confuse with the confocal metadata one)..

Considering how little it really does, maybe these arguments (pixelsizes and dwell time) should just be arguments of `save_tiff` instead.

Reply:
good point, I didn't think about the state of the pixelsize. And yes, I'd prefer these to be arguments of `save_tiff` rather than a separate class

Reply:
Agreed. I removed it and just made the resolution and time arguments for `save_tiff`.